File: src/main/java/com/demonwav/mcdev/buildsystem/maven/pom/Repositories.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2016 Kyle Wood (DemonWav)
 *
 * MIT License
 */
package com.demonwav.mcdev.buildsystem.maven.pom;

import com.intellij.util.xml.DomElement;
import com.intellij.util.xml.SubTagList;

import java.util.List;

public interface Repositories extends DomElement {

    @SubTagList("repository")
    List<Repository> getRepositories();

    Repository addRepository();
}

File: src/main/kotlin/com/demonwav/mcdev/buildsystem/maven/pom/Repositories.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2017 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.buildsystem.maven.pom

import com.intellij.util.xml.DomElement
import com.intellij.util.xml.SubTagList

interface Repositories : DomElement {

    @get:SubTagList("repository")
    val repositories: List<Repository>

    fun addRepository(): Repository
}
